src/common/multiline-ellipsis/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Adds an ellipsis to a el based on the height and width
+ * specified in CSS rules
+ */
+function truncate(el) {
+    if (!el) {
+        return;
+    }
+
+    const text = el.innerText;
+
+    if (this.isTextOverflowing(el)) {
+        const checkFunc = i => {
+            el.innerText = text.substring(0, i);
+            return this.isTextOverflowing(el) ? -1 : 0;
+        };
+        const len = this.binarySearch(text.length - 1, checkFunc);
```

Comment:
Seems a little strange to use `this` here. I think it would be better to just use the hoisted `binarySearch` function as is. This means that if the truncate function is bound for some reason it doesn't break.

Reply:
Done.